src/rust/plugin-execution-sidecar/src/work/plugin_work_processor.rs

```diff
@@ -52,20 +52,20 @@ pub trait PluginWorkProcessor {
 
     async fn get_work(
         &self,
-        config: &PluginExecutorConfig,
-        pwq_client: &mut PluginWorkQueueServiceClient,
+        plugin_id: Uuid,
+        pwq_client: &mut PluginWorkQueueClient,
     ) -> Result<Self::Work, PluginWorkProcessorError>;
 
     async fn process_job(
         &mut self,
-        config: &PluginExecutorConfig,
+        plugin_id: Uuid,
         work: ExecutionJob,
     ) -> Result<Self::ProducedMessage, PluginWorkProcessorError>;
 
     async fn ack_work(
         &self,
-        config: &PluginExecutorConfig,
-        pwq_client: &mut PluginWorkQueueServiceClient,
+        plugin_id: Uuid,
```

Comment:
If at some future time we find a need for more information we'll have two options

(1) create some configuration struct
(2) add a field to the method signature

It seemed to me here that the `ack_work` verb in this API can't do anything without a `plugin_id`, since we can't know which plugin we're acknowledging work _for_. Same for all the other verbs in this API. Since the `PluginExecutorConfig` only had one field--the `plugin_id` I think it made sense to lift this to an actual first-class entity in the API--it doesn't _add_ anything to hide the fact that there's a `plugin_id` here.